python/cudf/cudf/core/column/column.py

```diff
@@ -2142,11 +2138,19 @@ def as_column(
 
         if delayed_cast:
             data = data.astype(cudf.dtype(dtype))
+        return data
 
     elif (view := as_memoryview(arbitrary)) is not None:
         return as_column(
             np.asarray(view), dtype=dtype, nan_as_null=nan_as_null
         )
+    elif hasattr(arbitrary, "__array__"):
+        # e.g. test_cuda_array_interface_pytorch
```

Comment:
So PyTorch arrays support `__array__` but not `__array_interface__` or `__cuda_array_interface__`? Are there other types like that that we care about (based on what you've seen in our tests)?

Reply:
Correct AFAICT. This was the only special "array like" we test in cudf (I'm surprised this is supported)